src/main/java/jenkins/plugins/openstack/compute/JCloudsComputer.java

```diff
@@ -145,7 +145,7 @@ private void checkSlaveAfterTaskCompletion() {
 
         // If the retention time for this computer is zero, this means it
         // should not be re-used: mark the node as "pending delete".
-        if (getRetentionTime() == 0) {
+        if (getRetentionTime() == 0 && !(getOfflineCause() instanceof OfflineCause.UserCause)) {
```

Comment:
When user takes it to offline during an execution now it is his responsibility to click to 'schedule for termination' right? Otherwise the compute will be reserved undefinitely?

Reply:
Yes, same as with non-zero retention time.